src/openrct2/peep/Guest.cpp

```diff
@@ -3633,7 +3633,7 @@ static void peep_update_ride_leave_entrance_waypoints(Peep* peep, Ride* ride)
     Guard::Assert(!entranceLocation.isNull());
     uint8_t direction_entrance = entranceLocation.direction;
 
-    LocationXY16 waypoint;
+    CoordsXY waypoint;
     waypoint.x = ride->stations[peep->current_ride_station].Start.x * 32 + 16;
```

Comment:
Use ToTileCentre()

Reply:
I've done it, but note that it only sums the 16, but doesn't `floor`s the coords first.

Reply:
It doesn't need to - it multiplies by 32 first, so the value is a multiple of 32 already - already floored.